### ISSUE TITLE
Web cam

### ISSUE THREAD

**Issue by aliraza1616** [arunponnusamy/gender-detection-keras#3]
HI. i have checked Your code and its working fine on still. When i ran the model of Webcam, it always displayed man and women as man. can you please provide some help?

**Comment by arunponnusamy** [comment]
Hello @aliraza1616 , In this repository I have demonstrated how to train a neural network from scratch (without using transfer learning / fine tuning pre-trained model weights) for a task like 'gender detection'. The accuracy of the detection will be low mostly due to 
1. very small training dataset and
2. shallow / lightweight network architecture (in this case SmallerVGGNet).

 I am working on building more accurate gender detector. I will be adding it to [cvlib](http://cvlib.net) in the next release. Keep on eye on the [blog](http://arunponnusamy.com) for new posts. 

Regarding counting faces, you can just keep a counting variable inside the [loop](https://github.com/arunponnusamy/gender-detection-keras/blob/cb922b07c8189a967fbd8b24d12cdf9dd5846b41/detect_gender_webcam.py#L85)  and keep incrementing them whenever a face is detected and the control comes inside the loop.

**Comment by aliraza1616** [comment]
Hi.
I am trying to add images to data sets fkr its complete training.
Secondly i have to add counter too. Can you please add a line in code for me? I will be highly obliged if you can. As i am new to this field and i dont know much. Please if possible.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by aliraza1616** [comment]
Hi,

Can i run any Video in Gender recognition code? if yes kindly tell how to?
________________________________

**Comment by arunponnusamy** [comment]
Hello @aliraza1616 , you can replace the zero in `VideoCapture(0)` with path to video file name to work with video input. `VideoCapture('video_input.mp4')`
https://github.com/arunponnusamy/gender-detection-keras/blob/cb922b07c8189a967fbd8b24d12cdf9dd5846b41/detect_gender_webcam.py#L23

**Comment by aliraza1616** [comment]
Hi!!
I need t know that some of the faces are not detected by model. Like in attached pic. How to increase accuracy of detecting? Kindly reply for both the queries. Face count and face detection.

BR,
Ali

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by arunponnusamy** [comment]
Can you attach the image please ? 

Regarding face count, as I have mentioned earlier, you can just keep a counting variable inside the loop and keep incrementing them whenever a face is detected and the control comes inside the loop.
https://github.com/arunponnusamy/gender-detection-keras/blob/cb922b07c8189a967fbd8b24d12cdf9dd5846b41/detect_gender_webcam.py#L85

If you are new to Python, I would suggest to go through some introductory material on [realpython](http://realpython.com) or [codecademy](http://codecademy.com).

**Comment by aliraza1616** [comment]
Image is already attached bro.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by arunponnusamy** [comment]
Can you visit the current issue on [github](https://github.com/arunponnusamy/gender-detection-keras/issues/3) and attach the image ? replying to email directly with image doesn't actually help.

**Comment by nyck33** [comment]
@arunponnusamy thanks for a great contribution.  
do you think it's worth the time to use your model for transfer learning or do you think I should just start with a deeper Resnet or VGG16?

**Comment by arunponnusamy** [comment]
Hey @nyck33, If you are using ResNet / VGG16 I would suggest to start with ImageNet weights for transfer learning.

**Comment by nyck33** [comment]
@arunponnusamy Thanks for the reply and I'm very sorry for the bomb of questions.  
That''s a great idea but I'm also looking at VGGFace right now to use as the 'backbone'?

**Comment by arunponnusamy** [comment]
I haven't worked much with VGGFace but I guess you can do that too.
